src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -429,10 +429,19 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 		PlanState *planState = &(scanState->customScanState.ss.ps);
 
 		ExecuteMasterEvaluableFunctions(jobQuery, planState);
+		ExecuteTaskListFunctions(taskList, planState);
```

Comment:
I think we should generate tasks when we're ready to generate the correct number of tasks, which might only be at execution time (e.g. function in the partition column). I guess there is an important distinction between queries with a function call in the partition column, and queries with a constant in the partition column, but a function call in another column. In the former case we cannot generate the correct number of tasks at planning time (use `ExecuteMasterEvaluableFunctions(jobQuery, planState)`?), while in the latter case we can (use `ExecuteTaskListFunctions(taskList, planState)`?), though I don't know if we get enough performance benefit out of building the task list upfront to justify the additional function.

In any case we shouldn't do function evaluation twice, that would give unexpected results (e.g. `nextval` getting called twice).